src/google-analytics/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @implements {GoogleAnalyticsService}
+ * @memberof module:GoogleAnalytics
+ */
+class GoogleAnalyticsService {
+  constructor({ config }) {
+    this.analyticsId = config.GOOGLE_ANALYTICS_4_ID;
+
+    if (this.analyticsId) {
+      this.initializeService();
+    }
+  }
+
+  initializeService() {
+    global.googleAnalytics = global.googleAnalytics || [];
+    const { googleAnalytics } = global;
+
+    // If the snippet was invoked do nothing.
+    if (googleAnalytics.invoked) {
+      return;
+    }
+
+    // Invoked flag, to make sure the snippet
+    // is never invoked twice.
+    googleAnalytics.invoked = true;
+
+    googleAnalytics.load = (key, options) => {
+      const scriptSrc = document.createElement('script');
+      scriptSrc.type = 'text/javascript';
+      scriptSrc.async = true;
+      scriptSrc.src = `https://www.googletagmanager.com/gtag/js?id=${key}`;
+
+      const scriptGtag = document.createElement('script');
+      scriptGtag.innerHTML = `
+        window.dataLayer = window.dataLayer || [];
+        function gtag(){dataLayer.push(arguments);}
+        gtag('js', new Date());
+        gtag('config', '${key}');
+      `;
+
+      // Insert our scripts next to the first script element.
+      const first = document.getElementsByTagName('script')[0];
+      first.parentNode.insertBefore(scriptSrc, first);
+      first.parentNode.insertBefore(scriptGtag, first);
+      googleAnalytics._loadOptions = options; // eslint-disable-line no-underscore-dangle
+    };
+
+    // Load GoogleAnalytics with your key.
+    googleAnalytics.load(this.analyticsId);
+  }
+}
```

Comment:
Are we not going to implement artificial events such as [sendPageEvent()](https://github.com/openedx/frontend-platform/blob/master/src/analytics/SegmentAnalyticsService.js#L234-L241)?

I don't want to unwittingly extend scope, here.  It's just that I originally imagined a drop-in solution for replacing Segment with GA4.  But that's of course more complicated than just adding GA4 tags into MFEs that invoke this, which currently none do.

In other words, to use GA4 in MFEs with this new service, they still have to be forked or changed to do so.

Reply:
Yeah for this work we're focusing on 1:1 functionality with GA universal, not Segment. I believe 2U has work coming down later this year to extend the functionality closer to how Segment is working, but I don't know how far investigation has gone in that direction.